### PR TITLE
Update plot ci to return matplotlib plots objects

### LINKED ISSUE #136: Return lines and images from plot_ci method
to enable user to have more freedom in adjusting plots properties and to be consistent with geometry plotting methods, return matplotlib lines and images object created within plot_ci.

### LINKED ISSUE #138: Enable viewing methods full documentation in the online documentation of CUQIpy
Methods in the documentation is not clickable and their full documentation cannot be viewed. e.g. `plot_ci` in [Samples](https://cuqi-dtu.github.io/CUQIpy/api/_autosummary/cuqi.samples/cuqi.samples.Samples.html#cuqi.samples.Samples) documentation. The docstring for `plot_ci` has description of the parameters that does not show in the documentation.

### PR DESCRIPTION
**Pull request by amal-ghamdi**
closes #136 and closes #138

### CODE REVIEW

File: cuqi/samples/_samples.py [hunk 1]
```diff
@@ -408,16 +408,31 @@ def hist_chain(self,variable_indices,*args,**kwargs):
...
+            :class:`~matplotlib.lines.Line2D` object of the mean plot, 
+            :class:`~matplotlib.lines.Line2D` object of the exact value plot,
+            and :class:`~matplotlib.collections.PolyCollection`
+            or :class:`~matplotlib.container.ErrorbarContainer`
+            object of the ci envelope plot, respectively.
+
+            If 2D plots are generated, the list contains
+            :class:`~matplotlib.collections.PolyCollection` objects for
+            the mean, exact value, ci lower bound, ci upper bound, and the 
+            ci width, respectively.
```
Comment (nabriis) [inline cuqi/samples/_samples.py:435]:
```suggestion
            If 2D plots are generated, the list contains
            :class:`~matplotlib.collections.PolyCollection` or 
            :class:`~matplotlib.image.AxesImage` objects
            for the mean, exact value, ci lower bound, ci upper
            bound, and the ci width, respectively.
```

File: docs/_templates/autosummary/class.rst [hunk 2]
```diff
@@ -11,6 +11,7 @@
    .. rubric:: {{ _('Methods') }}
 
    .. autosummary::
```
Comment (nabriis) [inline docs/_templates/autosummary/class.rst:13]: Such a huge change for 2 lines of code changed :) Looks fantastic. Thanks for adding this!

File: tests/test_samples.py [hunk 3]
```diff
@@ -196,6 +197,39 @@ def test_plot_ci_par_func(is_par, plot_par, compute_on_par, geometry):
         samples.plot_ci(plot_par=plot_par)
 
 
+@pytest.mark.parametrize("geometry",
+                         [cuqi.geometry.Discrete(4),
+                          cuqi.geometry.KLExpansion(np.linspace(0, 1, 4)),
+                          cuqi.geometry.Continuous2D((2, 2))])
```
Comment (nabriis) [inline tests/test_samples.py:203]: Add test for Image2D geometry

**Review by nabriis — CHANGES_REQUESTED** [review]
This is really nice @amal-ghamdi. Thanks for adding the option to get back the matplotlib figure objects. Also thanks for adding the super nice feature to be able to inspect methods / attributes of classes in the docs! Really great.

File: tests/test_samples.py (hunk 3 above)
Comment (amal-ghamdi) [reply tests/test_samples.py]: Thank you @nabriis catching this case (Image2D) in the documentation and the test 👍. I added a test case for it.

File: docs/_templates/autosummary/class.rst (hunk 2 above)
Comment (amal-ghamdi) [reply docs/_templates/autosummary/class.rst:13]: :)

**Comment by amal-ghamdi** [comment]
Thank you @nabriis for taking time to review this PR!

File: cuqi/samples/_samples.py [hunk 4]
```diff
@@ -491,6 +516,14 @@ def plot_ci(self, percent=95, exact=None, *args, plot_envelope_kwargs=None, **kw
             else:
                 plt.legend([lmn[0], lci], ["Mean", "Credibility Interval"])
 
+            plotting_objects = [lmn, lex, lci]
+        
+        # Form a list of the matplotlib objects that were plotted
```
Comment (jakobsj) [inline cuqi/samples/_samples.py:521]: Please explain the `obj[0]`.

**Review by jakobsj — APPROVED** [review]
Looks great!  Just one tiny clarification to comment on, after that you can merge.

**Comment by amal-ghamdi** [comment]
Thank you @jakobsj for your review and helpful comment! I added some clarification on why we index using obj[0].

**Pull request merged**